llvm_jni/main.bal

```diff
@@ -4,10 +4,11 @@ public function main() {
     llvm:Context context = new;
     llvm:Builder builder = context.createBuilder();
     llvm:Module m = context.createModule();
-    llvm:Function mainFunction = m.addFunctionDefn("main", {returnType: "void", paramTypes: ["i64", "i1"]});
+    llvm:Function mainFunction = m.addFunctionDefn("main", {returnType: "i64", paramTypes: ["i64", "i1"]});
     llvm:BasicBlock initBlock = mainFunction.appendBasicBlock();
```

Comment:
We need a proper command-line driver program that takes a `.bal` file plus some command-line arguments and compiles it to a `.o` file. You should use Ballerina's built-in CLI handling capabilities like the nballerina module does.

Reply:
Fixed by modifying the driver to directly take the optional command line arguments to object file generation and changed the `print.llvm` to panic if asked to generate object file

Reply:
That's not what I'm looking for. I want a **separate** driver program in llvm_jni. Doesn't have to be fancy: just enough to talk a file and compile it to `.o`.

Reply:
For clarification what you are suggesting is move the parts that has to be different (`main.bal` and `nback.bal`) in to the llvm_jni (and sym link the reset of compiler in to llvm_jni because above 2 depend on them for development purposes) and at the build time (when are currently replace print.llvm with jni.llvm) replace those different files with the versions from jni.llvm

Reply:
I am suggesting a new `main.bal` file (completely separate from the current one) that provides a `main` function that uses `nback`, `front` and `jni.llvm` modules to compile a .bal file into a `.o` file. This may require refactoring/extending `nback` so that `main` has access to the llvm:Module and llvm:Context (e.g. we could make `buildModule` public). Please suggest something.

The reason is that a `main` that can compile all the way to `.o` is (eventually) going to have a ton of options (which will be passed to LLVM) that will be irrelevant for print.llvm. This `main` will also (eventually) be responsible for managing the jBallerina front-end.

Reply:
I think the easiest way will be to change the `compileFile` function in `main.bal` and `nback`'s `compileModule` functions. We will change `compileModule` so that it will return a `llvm:module` (similar to `front:loadModule` return a `bir:Module`) instead of taking care of output as well. Then `main.bal` will decided how to handle output. This way `main` will have direct access to all three `nback`, `front` and `jni.llvm`|`print.llvm`

Reply:
`compileModule` will then be doing the same thing that `buildModule` currently does. So we can just

- move `nback:compileModule` into `main.bal`
- make `nback:buildModule` public (and perhaps rename it)

Reply:
Fixed